docs/subnets/hello-world-precompile-tutorial.md

```diff
@@ -0,0 +1,953 @@
+
+```go
+// StatefulPrecompiledContract is the interface for executing a precompiled contract
+type StatefulPrecompiledContract interface {
+	// Run executes the precompiled contract.
+	Run(accessibleState PrecompileAccessibleState,
+	caller common.Address,
+	addr  common.Address,
+	input []byte,
+	suppliedGas uint64,
+	readOnly bool)
+	(ret []byte, remainingGas uint64, err error)
+}
+```
+
+A stateful precompile injects state access through the `PrecompileAccessibleState` interface to provide access to the EVM state including the ability to modify balances and read/write storage.
+
+This way we can provide even more customization of the EVM through Stateful Precompiles than we can with the original precompile interface!
+
+## Tutorial
+
+### Overview
+
+This is a brief overview of what this tutorial will cover.
+
+- Write a Solidity Interface
+- Generate the precompile template
+- Implement the precompile functions in Golang
+- Write and run tests
+
+Stateful precompiles are [alpha software](https://en.wikipedia.org/wiki/Software_release_life_cycle#Alpha). Build at your own risk.
+
+### Prerequisites
+
+This tutorial assumes familiarity with Golang and Javascript.
+
+Additionally, users should be deeply familiar with the EVM in order to understand its invariants since adding a Stateful Precompile modifies the EVM itself.
+
+Here are some recommended resources to learn the ins and outs of the EVM:
+
+- [The Ethereum Virtual Machine](https://github.com/ethereumbook/ethereumbook/blob/develop/13evm.asciidoc)
+- [Precompiles in Solidity](https://medium.com/@rbkhmrcr/precompiles-solidity-e5d29bd428c4)
+- [Deconstructing a Smart Contract](https://blog.openzeppelin.com/deconstructing-a-solidity-contract-part-i-introduction-832efd2d7737/)
+- [Layout of State Variables in Storage](https://docs.soliditylang.org/en/v0.8.10/internals/layout_in_storage.html)
+- [Layout in Memory](https://docs.soliditylang.org/en/v0.8.10/internals/layout_in_memory.html)
+- [Layout of Call Data](https://docs.soliditylang.org/en/v0.8.10/internals/layout_in_calldata.html)
+- [Contract ABI Specification](https://docs.soliditylang.org/en/v0.8.10/abi-spec.html)
+- [Precompiles in Solidity](https://medium.com/@rbkhmrcr/precompiles-solidity-e5d29bd428c4)
+- [Customizing the EVM with Stateful Precompiles](https://medium.com/avalancheavax/customizing-the-evm-with-stateful-precompiles-f44a34f39efd)
+
+Please install the following before getting started.
+
+First install the latest version of Go. Follow the instructions [here](https://go.dev/doc/install). You can verify by running `go version`.
+
+Set `$GOPATH` environment variable properly for Go to look for Go Workspaces. Please read [this](https://go.dev/doc/gopath_code) for details. You can verify by running `echo $GOPATH`.
+
+As a few things will be installed into `$GOPATH/bin`, please make sure that `$GOPATH/bin`is in your `$PATH`, otherwise, you may get error running the commands below.
```

Comment:
```suggestion
As a few things will be installed into `$GOPATH/bin`, please make sure that `$GOPATH/bin` is in your `$PATH`, otherwise, you may get error running the commands below.
```

Reply:
thanks